src/emulator/module/src/module.cpp

```diff
@@ -27,31 +27,25 @@ static std::mutex mutex;
 static NameSet logged;
 
 int unimplemented(const char *name) {
+#ifndef VERBOSE    
```

Comment:
I think this should be an `#if VERBOSE` and put a `#define VERBOSE 0` someplace so it can be easily disabled/enabled, especially since `VERBOSE` isn't a standard macro afaik.
Or better yet, don't use the preprocessor at all and just make a `constexpr` flag.

Reply:
VERBOSE afaik it's a common macro in big projects to enable extended logging.
Btw, "if VERBOSE" with VERBOSE set to 0 or "ifndef VERBOSE" have the same exact result.

Reply:
`#ifdef` checks whether it's **defined**, it doesn't care about the value, whereas `#if` actually checks if the **value** is 0 or not.
In your code you'd need to either define it, or comment out the define or undef it.

I didn't know `VERBOSE` was common, and even if so, you can't assume that everyone coming in to this project will assume it's there so still nice to have it nice somewhere, like the `LOG_*` flags.

Reply:
I'd like to know the rationale behind logging this more than once. I'm not against if if there's a reason, just that I'm not sure at this point what the reason is.

Reply:
As i explained on Discord, an application can have unwanted behaviours due to several calls to an unimplemented nid but you'll never know about how many times it got called due to it being logged once.